### DISCUSSION TITLE
App Registration Fees: Is there any other way?

### DISCUSSION THREAD

**Discussion by shenied-maldonado** [uprm-inso4116-2024-2025-s1/semester-project-SafeRUM#139] [Ideas]
Hey team! 

I've been researching ways to attach a third-party user authentication service for our app. As owned by Google, Firebase offers Firebase Authentication which includes features like email verification, password recovery, session management, and user data storage to ensure that sensitive information is securely handled. I had written up some issues to implement it for iOS and Android devices separately, but further research has shown me that that alternative requires us to pay app registration fees. Ideally, we'd probably like to take advantage of more platform-specific advantages that come along with registering our app. However, wanting to avoid that for now, I am proposing using **Firebase's web-based authentication** in the meantime. This will let us avoid platform-specific registration but includes some trade-offs in user experience and access to some native functionalities (i.e., push notifications, deep linking, etc.).

I explored using other third-party authentication libraries like Auth0, but we'd still be asked for an Apple Bundle ID or an Android package name that we'd only be able to get by registering our app. I don't want to do anything with @YanehRuiz or @evnd04 or any other stakeholder's approval, so please let me know what you think! If anyone has any other ideas, I'm open to suggestions!

**Comment by YanehRuiz** [comment]
Hello!

Could you perhaps provide screenshots of the cost prompt? Firebase has a Spark plan, which is free, if we don't apply SMS authentication, however, it should allow for google and other authentication methods!

**Comment by shenied-maldonado** [comment]
Firebase hasn't required any cost in any process I've explored, but it does require our mobile app to be registered with either Apple or Android. What I'm proposing is to attach Firebase's web authentication to SafeRUM to avoid having to pay registration fees for either developer program. 

Here is what registering our app on Firebase's console looks like (a necessary step before configuring its authentication services with SafeRUM). 

![Screenshot 2024-09-15 at 2 27 48 PM](https://github.com/user-attachments/assets/e3af4e75-c2e5-42ca-97d3-5ce5613d629d)

![Screenshot 2024-09-15 at 2 27 59 PM](https://github.com/user-attachments/assets/2a13e0c7-a778-4748-8dba-f5e3dda53c20)

The cost issue comes up with the Apple bundle ID and Android package name Firebase requirements. While we can entertain the idea of developing an iOS app, enrolling in the Apple Developer Program is an obligation for distribution of our app and acquisition of an Apple bundle ID. We can similarly develop an app for Android devices, but inclusion on the Google Play Store and access to our would-be Android package name requires app registration. 

According to their [page](https://developer.apple.com/support/enrollment/), "The Apple Developer Program annual fee is 99 USD and the Apple Developer Enterprise Program annual fee is 299 USD, in local currency where available. Prices may vary by region and are listed in local currency during the enrollment process." In Android's case, there seems to be a single payment of $25 USD for publishing our app on the Google Play store and, thus, getting assigned an Android package name. 

![Screenshot 2024-09-15 at 2 38 40 PM](https://github.com/user-attachments/assets/cca88da3-a701-44af-a409-fbdef8f1ba8a)

I'm not sure if there is a way to bypass these fees for distribution purposes but, in terms of being able to apply Firebase authentication, I can attach its web SDK. It's not ideal for our app, but it's the only alternative I can come up with right now. I'll do more research and wait for feedback as well as further instructions in the meantime.
